File: src/workflows/load_generator/load_generator_activities.py
"""Activities for the load generator workflow."""

from pydantic import BaseModel
from temporalio import activity


class NoopActivityInput(BaseModel):
    """Input for the noop activity."""

    message: str = "ping"


class NoopActivityOutput(BaseModel):
    """Output for the noop activity."""

    message: str = "pong"


@activity.defn
async def noop_activity(input: NoopActivityInput) -> NoopActivityOutput:
    """A simple no-op activity that just returns a value."""
    return NoopActivityOutput(message=f"processed: {input.message}")

File: src/workflows/load_generator/load_generator_activities.ts
import { Context } from '@temporalio/activity';

export interface NoopActivityInput {
  message: string;
}

export interface NoopActivityOutput {
  message: string;
}

function getLogger(): Pick<Console, 'info'> {
  try {
    return Context.current().log;
  } catch {
    return console;
  }
}

export async function noopActivity(input: NoopActivityInput): Promise<NoopActivityOutput> {
  getLogger().info('Processing noop activity', { message: input.message });

  return {
    message: `processed: ${input.message}`,
  };
}
